lib/middlewares/validateSignupData.js

```diff
@@ -0,0 +1,38 @@
+import Schema from 'validate'
+
+const signUpData = {
+  firstName: {
+    type: String,
+    required: true,
+    length: { min: 3, max: 55 },
+    message: 'please povide a valid first name',
+  },
+  lastName: {
+    type: String,
+    required: true,
+    length: { min: 3, max: 55 },
+    message: 'please povide a valid last name',
+  },
+  email: {
+    type: String,
+    required: true,
+    match: /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/,
+    message: 'please provide a valid email',
+  },
+  password: {
+    type: String,
+    required: true,
+    length: { min: 6 },
+  },
+}
+
+const validateSignupData = (req, res, next) => {
+  const userData = new Schema(signUpData)
+  const errors = userData.validate(req.body)
+  if (errors.length) {
+    return res.status(400).json({ message: errors[0].message })
```

Comment:
We can have a uniform error hanlder for all the errors, so this allows us to pass errors to function next()
e.g ```errors.length ? next({ message: errors[0].message }) : next()```